### PR TITLE
Fetch and set up secret for skaffold-metrics project

### PR DESCRIPTION
**Pull request by tejal29**
In this PR
1) Use secret from newly created  project
2) Drop the 2 weeks validity as longer keys exist.

No code change required for instrumentation as project id is [derived](https://github.com/GoogleContainerTools/skaffold/blob/master/pkg/skaffold/instrumentation/export.go#L122) from the secret itself.

Earlier dashboard can be used as k8s-skaffold is defines as scoped project https://cloud.google.com/monitoring/settings#concept-scope

### CODE REVIEW

File: deploy/setup-secret.sh [hunk 1]
```diff
@@ -30,10 +31,10 @@ done
 
 function download_existing_key() {
   # Download a valid key created within the past two weeks.
-  KEY_IDS=$(gcloud iam service-accounts keys list --iam-account=metrics-writer@k8s-skaffold.iam.gserviceaccount.com --project=k8s-skaffold --managed-by=user --filter="validAfterTime>-P2W" --format="value(name)")
+  KEY_IDS=$(gcloud iam service-accounts keys list --iam-account=metrics-writer@${METRICS_PROJECT_ID}.iam.gserviceaccount.com --project=${METRICS_PROJECT_ID} --managed-by=user --format="value(name)")
   while read -r KEY_ID
   do
-    if gsutil cp gs://${BUCKET_ID}/${KEY_ID}.json ${KEY_FILE}; then
+    if gsutil cp gs://${BUCKET_ID}/${LATEST_GCS_PATH}.json ${KEY_FILE}; then
```
Comment (aaron-prindle) [inline deploy/setup-secret.sh:37]: LATEST_GCS_PATH=keys.json so I think this will be `keys.json`.json

File: deploy/setup-secret.sh [hunk 2]
```diff
@@ -30,10 +31,10 @@ done
 
 function download_existing_key() {
   # Download a valid key created within the past two weeks.
-  KEY_IDS=$(gcloud iam service-accounts keys list --iam-account=metrics-writer@k8s-skaffold.iam.gserviceaccount.com --project=k8s-skaffold --managed-by=user --filter="validAfterTime>-P2W" --format="value(name)")
+  KEY_IDS=$(gcloud iam service-accounts keys list --iam-account=metrics-writer@${METRICS_PROJECT_ID}.iam.gserviceaccount.com --project=${METRICS_PROJECT_ID} --managed-by=user --format="value(name)")
```
Comment (aaron-prindle) [inline deploy/setup-secret.sh:34]: nit: is this still needed? `--filter="validAfterTime>-P2W"` I'm guessing that w/ the new project and expiration date it isn't, just verifying

File: deploy/setup-secret.sh (hunk 1 above)
Comment (tejal29) [reply deploy/setup-secret.sh]: fixed.

File: deploy/setup-secret.sh (hunk 2 above)
Comment (tejal29) [reply deploy/setup-secret.sh:34]: yeah. new keys live longer and hence we don't need that check.

**Review by aaron-prindle — APPROVED** [review]
LGTM

**Pull request merged**